website/content/en/docs/best-practices/multi-tenancy.md

```diff
@@ -0,0 +1,162 @@
+follow the least-privilege principle as well implement deny-all-allow-some rules.
+
+### Managing Resources
+
+#### Resource Requests
+
+Create resource requests for CPU and memory:
+
+* for your Operator's manager `Deployment` (See `config/manager/manager.yaml`)
+* for any `Pods` and `Deployments` which are created by your Operator
+
+```yml
+  ...
+  # TODO(user): Configure the resources accordingly based on the project requirements.
+  # More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+  resources:
+    requests:
+      cpu: 10m
+      memory: 64Mi
+    ... 
+```
+
+In order to prevent tenants from consuming all the resources on a cluster or blocking other workloads 
+from being scheduled, many production clusters will define [ResourceQuota][k8s-resources-quotas] configuration.
+[ResourceQuota][k8s-resources-quotas] configuration in a namespace will cause all pods without 
+[Requests and limits][k8s-requests-limits] for `CPU` and `Memory` to be rejected.
+
+Because OLM-managed Operators do not run on Cluster infra nodes, they will consume paid CPU cores. 
+Users will naturally want to limit the maximum amount of resources these non-workload components can take up. 
+For this reason, if you are integrating with [OLM][olm], you should, **always ship your operator `Deployment` with resource requests amounts for CPU and memory**.
+
+The [ResourceQuota][k8s-resources-quotas] configuration also is applied to tenant workloads that are managed by your operator. Cluster administrators will typically set
+a [ResourceQuota][k8s-resources-quotas] for each tenant's namespace/project as part of the onBoarding. If your operator creates `Pods/Deployments` inside
+the tenant namespace which do not specify at least resource requests for `CPU` and `Memory`, those `Pods/Deployments` will fail to schedule and tenants
+will not be able to use your operator. Therefore, your operator should **always apply at least resource requests for CPU and
+memory** for `Pods/Deployments` that it creates as part of reconciliation. Ideally your operator also applies resource 
+limits to those `Pods/Deployments`. 
+
+#### General guidelines
+
+**Note that pretty much all controllers consume:**
+- `CPU`: largely based on the number of reconciliations they perform, which are generally related to event activity for resources they're watching.
+- `Memory`: largely based on the number of primary resources that exist (multiplied by some factor based on the number of operand resources they need to watch as a result) via informer caches.
+
+Operator authors cannot anticipate how they scale in usage in all customer environments, so setting one-size-fits-all 
+that fits well for all scenarios is not possible then, clusters administrators must be allowed to change the values 
+in order to respond to changes in their clusters’ needs, such as by resizing control plane nodes to provide more resources.
```

Comment:
"clusters administrators must be allowed to change the values in order to respond to changes in their clusters’ needs, such as by resizing control plane nodes to provide more resources."
It is unclear to me (the reader) whether this is related to the operator or the operands.
As you know for the operator this can be done at the subscription level (mentioned later) and as far as I know there is nothing required from the operator author for that.
For the operands that's where work by the operator author is required, depending on the type of operand the authors should make it possible to configure the resources either through the custom resource created by the user (example Tekton) or through a ComponentConfig (example Prometheus)

Reply:
"such as by resizing control plane nodes to provide more resources" I think that this should be removed. The operator authors have no influence on that

Reply:
> "clusters administrators must be allowed to change the values in order to respond to changes in their clusters’ needs, such as by resizing control plane nodes to provide more resources."
It is unclear to me (the reader) whether this is related to the operator or the operands.

That is for any POD, so in this for both Operator and Operands. 

> As you know for the operator this can be done at the subscription level (mentioned later) and as far as I know there is nothing required from the operator author for that.

Any operator consumer can change the manager manifests, in OLM it is recommended via subscription to keep the change persistent.  However, note that the Operator does not mandatorily integrate the project with OLM. This doc is in general. 
For any other pod created/managed by the Operator it can be achieved by letting the consumer change the values via crd/cr specs or env vars. then, depends on how the solution will be designed. 

> For the operands that's where work by the operator author is required, depending on the type of operand the authors should make it possible to configure the resources either through the custom resource created by the user (example Tekton) or through a ComponentConfig (example Prometheus)

> ComponentConfig to controller-runtime to allow controller authors to make go types backed by apimachinery to unmarshal and configure the ctrl.Manager{}
from: https://github.com/kubernetes-sigs/controller-runtime/blob/master/designs/component-config.md

ComponentConfig is designed to config the Operator/manager and not Operands at all. 
it does **not** support changing the resources, and its idea is to configure the [manager options](https://pkg.go.dev/k8s.io/kubernetes/cmd/kube-controller-manager/app/options). 
We do have no resources in the manager options. 
I could not find the component config usage in: https://github.com/prometheus-operator/prometheus-operator
  
> "such as by resizing control plane nodes to provide more resources" I think that this should be removed. The operator authors have no influence on that

The goal here is clarified to the operator authors that cluster administrators must be allowed to resizing control plane nodes to provide more resources.
operator authors are able to develop solutions that allow or not it.

Reply:
We changed here as well. So, closing as sorted out and let's see the second round :-)